### PR TITLE
SDIT-2089 Initial resource for contact migration

### PR DESCRIPTION
**Pull request by andymarke**
Added call to get person details and improve test performance

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonMigrationService.kt [hunk 1]
```diff
@@ -34,7 +36,9 @@ class ContactPersonMigrationService(
     pageSize = pageSize,
   )
 
-  override suspend fun migrateNomisEntity(context: MigrationContext<PersonIdResponse>) = TODO()
+  override suspend fun migrateNomisEntity(context: MigrationContext<PersonIdResponse>) {
+    nomisApiService.getPerson(nomisPersonId = context.body.personId)
+  }
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonMigrationService.kt:41]: TODO - obvious actually migrate the record

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/history/MigrationMapping.kt [hunk 2]
```diff
@@ -51,20 +51,18 @@ abstract class MigrationMapping<MAPPING : Any>(
   open suspend fun createMapping(
     mapping: MAPPING,
     errorJavaClass: ParameterizedTypeReference<DuplicateErrorResponse<MAPPING>>,
-  ): CreateMappingResult<MAPPING> {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/history/MigrationMapping.kt:54]: ktlint only

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/MigrationService.kt [hunk 3]
```diff
@@ -18,6 +18,7 @@ abstract class MigrationService<FILTER : Any, NOMIS_ID : Any, MAPPING : Any>(
   private val pageSize: Long,
   private val completeCheckDelaySeconds: Int,
   private val completeCheckCount: Int,
+  private val completeCheckRetrySeconds: Int = 1,
 ) {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/MigrationService.kt:22]: made this configurable but default to old value. Set this to zero speeds up Integration tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/LocalStackContainer.kt [hunk 4]
```diff
@@ -36,7 +37,7 @@ object LocalStackContainer {
         Wait.forLogMessage(".*Ready.*", 1),
       )
       start()
-      followOutput(logConsumer)
+      followOutput(logConsumer, OutputFrame.OutputType.STDERR, OutputFrame.OutputType.END)
     }
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/LocalStackContainer.kt:41]: reduce noise in logs; without this we get useless debug type logs from SQS

File: src/test/resources/application-test.yml [hunk 5]
```diff
@@ -75,7 +75,8 @@ courtsentencing:
   has-migrated-data: true
 contactperson:
    complete-check:
-     delay-seconds: 1
+     delay-seconds: 0
+     retry-seconds: 0
      count: 1
```
Comment (andymarke) [inline src/test/resources/application-test.yml:80]: experiment to speed up test - which seems to work since we only migrate 2 contacts in our test

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**